tools/swift-machine-learning-api/codes/trainer/launch_demo.py

```diff
@@ -0,0 +1,722 @@
+        model = CannedModel(
+            model_name=hparams.name,
+            feature_columns=feature_cols,
+            deep_columns=deep_cols,
+            hidden_units=hidden_units,
+            n_classes=hparams.n_classes,
+            linear_optimizer=linear_optimizer,
+            dnn_optimizer=deep_optimizer,
+            activation_fn=hparams.activation_fn,
+            dropout=hparams.dropout,
+            batch_norm=hparams.batch_norm,
+            config=config)
+    else:
+        model = CustomModel(
+            model_name=hparams.name,
+            batch_size=hparams.batch_size,
+            optimizer=poly_optimizer,
+            model_dir=hparams.job_dir,
+            config=config,
+            feature_names=feature_names,
+            learning_rate=hparams.learning_rate)
+
+    def mean_acc_metric(labels, predictions):
+        """
+        Defining mean per class accuracy metric
+        Arguments:
+            labels : labels of the data
+            predictions : prediction of the model
+        Returns: function defining mean per class accuracy metric
+        """
+        return mean_acc(labels, predictions, hparams.n_classes)
+
+    estimator = model.build_model()
+    if data.task_type == 'classification' and hparams.n_classes == 2:
+        estimator = tf.contrib.estimator.add_metrics(estimator, my_auc)
+    elif hparams.n_classes > 2:
+        estimator = tf.contrib.estimator.add_metrics(
+            estimator, mean_acc_metric)
+    else:
+        estimator = tf.contrib.estimator.add_metrics(estimator, rmse)
+        estimator = tf.contrib.estimator.add_metrics(estimator, mar)
+
+    if hparams.early_stopping:
+        old_loss = np.inf
+        for _ in range(hparams.eval_times):
+            estimator.train(input_fn=train_input,
+                            steps=hparams.train_steps//hparams.eval_times)
+            output = estimator.evaluate(
+                input_fn=eval_input, steps=hparams.eval_steps)
+            loss = output['loss']
+            if loss >= old_loss:
+                tf.logging.info(
+                    'EARLY STOPPING....... LOSS SATURATED AT : %s', loss)
+                break
+            else:
+                old_loss = loss
+
+    else:
+        train_spec = tf.estimator.TrainSpec(
+            train_input,
+            hparams.train_steps)
+
+        eval_spec = tf.estimator.EvalSpec(
+            eval_input,
+            hparams.eval_steps,
+            throttle_secs=hparams.eval_freq)
+        tf.estimator.train_and_evaluate(estimator, train_spec, eval_spec)
+
+    if config_obj.get_is_chief():
+        estimator.export_savedmodel(
+            hparams.export_dir,
+            serving_functions[hparams.export_format],
+            assets_extra={
+                'lime_explainer': '/tmp/lime_explainer'},
+            strip_default_attrs=False)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
```

Comment:
This is a lot of parameters. Consider refactoring this into a config so CLI invocation would be just passing a reference to the config. This will make it easier for the user to have consistent invocations.

Reply:
The launch_demo is not supposed by user directly. It is actually trigged by a post request with parameter passed in the body. Therefore, CLI invocation is not needed.